app/scripts/metamask-controller.js

```diff
@@ -852,8 +855,83 @@ export default class MetamaskController extends EventEmitter {
             saveState: async () => {
               await this.keyringController.persistAllKeyrings();
             },
-            removeAccount: async (address) => {
-              await this.removeAccount(address);
+            addAccount: async (_address, origin, handleUserInput) => {
+              const { id: addAccountApprovalId } =
+                this.approvalController.startFlow();
```

Comment:
As we're starting a flow, it's best practice to use a `try finally` block to ensure it gets ended no matter what otherwise we have an infinite spinner in the notification.

The approval request created below for example will throw if rejected which, looking at the code, I think would cause this scenario.

Reply:
👍  resolved in [this commit](https://github.com/MetaMask/metamask-extension/pull/20684/commits/a06f752e23a61a4ec3cc8060c965df1e4f25c83b)